### PR TITLE
[CARBONDATA-2499][Test] Validate the visible/invisible status of datamap

### PR DESCRIPTION
**Pull request by xubo245**
This jira https://issues.apache.org/jira/browse/CARBONDATA-2380 not check the visible/invisible status of datamap.  There will not thro exception when the related code changed and affect visible/invisible status of datamap.  So we should add some test case for this function.

When I develop https://github.com/apache/carbondata/pull/2290, carbon don't throw exception when I change the org.apache.carbondata.core.datamap.DataMapStoreManager#getAllVisibleDataMap
Be sure to do all of the following checklist to help us incorporate 
your contribution quickly and easily:

 - [x] Any interfaces changed?
 No
 - [x] Any backward compatibility impacted?
 No
 - [x] Document update required?
No
 - [x] Testing done
        Please provide details on 
        - Whether new unit test cases have been added or why no new tests are required?
        - How it is tested? Please attach test report.
        - Is it a performance related change? Please attach the performance test report.
        - Any additional information to help reviewers in testing this change.
       add some test cases
 - [x] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA. 
No

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/4840/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/5018/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/4851/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/6020/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/5030/

File: integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/datamap/CGDataMapTestCase.scala [hunk 1]
```diff
@@ -420,6 +419,63 @@ class CGDataMapTestCase extends QueryTest with BeforeAndAfterAll {
       sql("select * from normal_test where name='n502670' and city='c2670'"))
   }
 
+  test("test cg datamap and validate the visible and invisible status of datamap ") {
```
Comment (xuchuanyin) [inline integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/datamap/CGDataMapTestCase.scala]: The above testcase can be removed.

File: integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/datamap/CGDataMapTestCase.scala [hunk 2]
```diff
@@ -420,6 +419,63 @@ class CGDataMapTestCase extends QueryTest with BeforeAndAfterAll {
...
+    sql(
+      s"""
+         | CREATE DATAMAP ggdatamap2 ON TABLE $tableName
+         | USING '${classOf[CGDataMapFactory].getName}'
+         | DMPROPERTIES('index_columns'='city')
+       """.stripMargin)
+    sql(s"LOAD DATA LOCAL INPATH '$file2' INTO TABLE $tableName OPTIONS('header'='false')")
+
+    val df1 = sql(s"EXPLAIN EXTENDED SELECT * FROM $tableName WHERE name='n502670'").collect()
+    assertResult(
```
Comment (xuchuanyin) [inline integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/datamap/CGDataMapTestCase.scala]: I think we can simply verify the result by judging whether the output of explain contains  `ggdatamap1`.

File: integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/datamap/FGDataMapTestCase.scala [hunk 3]
```diff
@@ -481,6 +479,62 @@ class FGDataMapTestCase extends QueryTest with BeforeAndAfterAll {
       sql("select * from normal_test where name='n502670' and city='c2670'"))
   }
 
+  test("test fg datamap and validate the visible and invisible status of datamap ") {
```
Comment (xuchuanyin) [inline integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/datamap/FGDataMapTestCase.scala]: I think you can add a new testcaseSuite  to test the visibility of datamap and it will eliminate some duplicate code.

Besides, I've noticed that you pointed that the invisibility for lucence datamap may cause failure.  Better to add that test in the same testcaseSuite.

File: integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/datamap/CGDataMapTestCase.scala (hunk 2 above)
Comment (xubo245) [reply integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/datamap/CGDataMapTestCase.scala]: Can you confirm it is correct if change it ontains ggdatamap1?

File: integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/datamap/FGDataMapTestCase.scala (hunk 3 above)
Comment (xubo245) [reply integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/datamap/FGDataMapTestCase.scala]: lucene datamap don't support invisibility feature, please refer:https://issues.apache.org/jira/browse/CARBONDATA-2502

File: integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/datamap/CGDataMapTestCase.scala (hunk 1 above)
Comment (xubo245) [reply integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/datamap/CGDataMapTestCase.scala]: Ok, I will add some validate in old test case

File: integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/datamap/FGDataMapTestCase.scala (hunk 3 above)
Comment (xubo245) [reply integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/datamap/FGDataMapTestCase.scala]: Jacky said there are many test case suite, it shouldn't add more test case suite if there exists related test case suite

File: integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/datamap/CGDataMapTestCase.scala (hunk 2 above)
Comment (xubo245) [reply integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/datamap/CGDataMapTestCase.scala]: I changed @xuchuanyin

File: integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/datamap/CGDataMapTestCase.scala (hunk 1 above)
Comment (xubo245) [reply integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/datamap/CGDataMapTestCase.scala]: ok, I removed。 and change old test case

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/6110/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/4948/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/5104/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/5105/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/4949/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/6112/

**Comment by xubo245** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/4953/

**Comment by xubo245** [comment]
@xuchuanyin CI pass

**Comment by xuchuanyin** [comment]
LGTM
